### ISSUE TITLE
Cannot estimate density for DM particles using add_sph_fields

### ISSUE THREAD

**Issue by jzuhone** [yt-project/yt#2326]
### Bug report

**Bug summary**

The `add_sph_fields` method for stream particle datasets is sometimes unable to generate a density field.

**Code for reproduction**

This can be triggered using the `ArepoBullet/snapshot_150.hdf5` dataset from http://yt-project.org/data:

```python
import yt

ds = yt.load("snapshot_154.hdf5")

ad = ds.all_data()
fields = ['particle_mass'] + [f'particle_position_{ax}' for ax in 'xyz']
data = {field: ad['PartType1', field] for field in fields}
ds_dm = yt.load_particles(data, data_source=ad)
ds_dm.add_sph_fields()
```

**Actual outcome**

```
yt : [INFO     ] 2019-09-11 18:18:28,016 Parameters: current_time              = 1.5780129111809016
yt : [INFO     ] 2019-09-11 18:18:28,017 Parameters: domain_dimensions         = [1 1 1]
yt : [INFO     ] 2019-09-11 18:18:28,018 Parameters: domain_left_edge          = [0. 0. 0.]
yt : [INFO     ] 2019-09-11 18:18:28,019 Parameters: domain_right_edge         = [40000. 40000. 40000.]
yt : [INFO     ] 2019-09-11 18:18:28,019 Parameters: cosmological_simulation   = 0.0
yt : [INFO     ] 2019-09-11 18:18:28,020 Generating SPH fields
yt : [INFO     ] 2019-09-11 18:18:28,021 Allocating for 1.000e+07 particles
yt : [INFO     ] 2019-09-11 18:18:44,383 Generating field ('io','smoothing_length')
Generate smoothing length: 10000001it [01:12, 137550.63it/s]                             
yt : [INFO     ] 2019-09-11 18:19:57,346 Generating field ('io','density')
Estimating density:  12%|█▏        | 1245185/10000000 [00:18<02:08, 68348.70it/s]
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-0be4329a5730> in <module>()
      3 data = {field: ad['PartType1', field] for field in fields}
      4 ds_dm = yt.load_particles(data, data_source=ad)
----> 5 ds_dm.add_sph_fields()

/pleiades/home6/jzuhone/src/yt/yt/frontends/stream/data_structures.py in add_sph_fields(self, n_neighbors, kernel, sph_ptype)
   1193             dens = estimate_density(
   1194                 pos[kdtree.idx], mass[kdtree.idx], hsml[kdtree.idx],
-> 1195                 kdtree, kernel_name=kernel,
   1196             )
   1197             dens = dens[order]

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.estimate_density()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_neighbors_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.find_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/particle_kdtree_tools.pyx in yt.utilities.lib.particle_kdtree_tools.process_node_points_ball()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/bounded_priority_queue.pyx in yt.utilities.lib.bounded_priority_queue.NeighborList.add_pid()

/pleiades/home6/jzuhone/src/yt/yt/utilities/lib/bounded_priority_queue.pyx in yt.utilities.lib.bounded_priority_queue.NeighborList._extend()

AttributeError: 'yt.utilities.lib.bounded_priority_queue._memoryvie' object has no attribute 'resize'
```

**Version Information**

  * Operating System: macOS and Linux
  * Python Version: 3.6, 3.7
  * yt version: 4.0.dev0

**Comment by jzuhone** [comment]
Very obscure error. I've looked at the Cython code but I have no idea what is going on here.

**Comment by qobilidop** [comment]
Looks like this is the part of code I touched recently. I'll take a look tomorrow.

**Comment by qobilidop** [comment]
Closed by #2328.
